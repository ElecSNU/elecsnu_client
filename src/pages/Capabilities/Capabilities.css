#capabilities-page {
    height: 100%;
    width: 100%;
    display: grid;
    place-items: center;
    overflow-y: scroll;
}

.capabilities-container {
    height: 80vh;
    margin: 6.825vh 0;
    width: 85%;
    padding: 2% 2.5%;
    display: flex;
    flex-direction: column;
    justify-content: space-evenly;
    align-items: center;
}

#capabilities-page .main-heading {
    font-size: 3.5rem;
}

.capabilities-links {
    width: 100%;
    display: flex;
    justify-content: space-between;
    align-items: center;
}

.capability-link {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
}

.capability-link .img-container {
    padding: 2.25rem;
    margin-bottom: 1.5rem;
    background-color: #004aad;
    border-radius: 100rem;
}

.capability-link img {
    margin: 0;
    width: 7.5rem;
}

.capability {
    height: 90%;
    display: flex;
    margin: 0;
    flex-wrap: wrap;
    align-items: center;
    justify-content: space-between;
    padding: 2.5%;
}

.capability-headings {
    flex: 1;
    width: 22.5%;
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    justify-content: space-evenly;
}

.capability-points {
    flex: 3;
    width: 72.5%;
    display: flex;
    flex-direction: column;
    align-items: flex-end;
    justify-content: space-between;
    list-style: none;
}

.capability-points li {
    margin-bottom: 1%;
    text-align: right;
}

.capability-points p {
    text-align: right;
}
