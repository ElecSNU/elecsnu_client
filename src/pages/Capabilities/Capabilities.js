import React from 'react';
import { Link } from 'react-router-dom';

import './Capabilities.css';

const Capabilities = () => {
    const capabilities = [
        {
            id: 'integrity',
            heading: 'Voting Integrity',
            subheading:
                'Trust the results of your votes and elections',
            points: [
                {
                    heading: 'One-to-One Voting',
                    text:
                        'Ensure voters are not able to cast multiple votes on the same ballot',
                },
                {
                    heading: 'Voter Authentication',
                    text:
                        'The voters are authenticated using Machine learning techniques like face detection and face recognition',
                },
            ],
        },
        {
            id: 'security',
            heading: 'Security',
            subheading:
                "Ensure the election's security and peace of mind.",
            points: [
                {
                    heading: 'Anonymity of votes',
                    text:
                        'Even the admin will not be able to see who voted for whom. Thus maintaining the anonymity of votes.',
                },
                {
                    heading: 'Voter Login',
                    text:
                        'Face detection technique is used to make sure only one person logs in while voting. Face recognition is also carried out to ensure that the person casting the vote and the one logged in are the same ',
                },
            ],
        },
        {
            id: 'admin',
            heading: 'Admin Experience',
            subheading:
                'Manage the voting process with ease',
            points: [
                {
                    heading: 'Voting Platform',
                    text:
                        'Gain control over your voting events. See the results analysis and everything at one place',
                },
                {
                    heading: 'Vote Management Services',
                    text:
                        'The admin can create groups of voters for different elections like club elections, student council elections, etc.',
                },
                {
                    heading: 'Voter Upload Modules',
                    text:
                        'Our voter list upload modules let you manage the list of eligible voters, their details, and login information.',
                },
                {
                    heading: 'Vote Scheduling Options',
                    text:
                        'Set elections to open and close at a preferred date and time',
                },
            ],
        },
        {
            id: 'voter',
            heading: 'Voter Experience',
            subheading:
                'Offer voters a straightforward, pleasant experience',
            points: [
                {
                    heading: 'Voter Notifications',
                    text:
                        'Even the admin will not be able to see who voted for whom. Thus maintaining the anonymity of votes.',
                },
                {
                    heading:
                        'Effective and User Friendly UI',
                    text:
                        'Providing voters an alternate login option using their college ID cards which is a unique-to-the-voter link that bypasses the traditional login page.',
                },
                {
                    heading: 'Candidate Manifestos',
                    text:
                        'There is an option to view the manifestos of all the candidates so that voters can make a more informed decision',
                },
                {
                    heading: 'Social Media',
                    text:
                        'In order to increase the voter turnout,we provide an option for the people who voted to upload a picture on their social media',
                },
            ],
        },
    ];

    return (
        <div id='capabilities'>
            <div className='capabilities-container background-light-translucent border-radius-15 box-shadow'>
                <h1 className='heading-text main-heading foreground-back capital-text center-text'>
                    Capabilities
                </h1>
                <div className='capabilities-links'>
                    {capabilities.map(
                        (capability, index) => (
                            <a
                                className='capability-link'
                                key={index}
                                href={`#${capability.id}`}
                            >
                                <img src='' alt='' />
                                <h1 className='heading-text foreground-accent1 capital-text center-text'>
                                    {capability.heading}
                                </h1>
                            </a>
                        )
                    )}
                </div>
            </div>
            {capabilities.map((capability, index) => (
                <>
                    <div
                        id={capability.id}
                        className={`capabilities-container border-radius-15 box-shadow capability foreground-${
                            index % 2 !== 0
                                ? 'back'
                                : 'light'
                        } background-${
                            index % 2 !== 0
                                ? 'light-translucent'
                                : '1-translucent'
                        }`}
                        key={index}
                    >
                        <div className='capability-headings'>
                            <h1 className='heading-text capital-text'>
                                {capability.heading}
                            </h1>
                            <h2 className='subheading-text'>
                                {capability.subheading}
                            </h2>
                        </div>
                        <ul className='capability-points'>
                            {capability.points.map(
                                (capabilityPoint, i) => (
                                    <li key={i}>
                                        <div className='subheading-text'>
                                            {
                                                capabilityPoint.heading
                                            }
                                        </div>
                                        <p className='normal-text'>
                                            {
                                                capabilityPoint.text
                                            }
                                        </p>
                                    </li>
                                )
                            )}
                        </ul>
                    </div>
                    <br />
                    <br />
                </>
            ))}
            <div style={{ opacity: '0' }}>.</div>
        </div>
    );
};

export default Capabilities;
