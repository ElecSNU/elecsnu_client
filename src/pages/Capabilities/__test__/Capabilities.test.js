import { render, screen, fireEvent, cleanup } from '@testing-library/react';
    import ReactDOM from 'react-dom'
    // import React , { useState }from 'react'
    import renderer from 'react-test-renderer'
    // import DesignSide from './../DesignSide'
    import Capabilities from './../Capabilities'

    test('Tests the rendering of Capabilities', () => {
        const div = document.createElement("div")
        const tree = renderer.create(<Capabilities></Capabilities>).toJSON();
        expect(tree).toMatchSnapshot();
       });