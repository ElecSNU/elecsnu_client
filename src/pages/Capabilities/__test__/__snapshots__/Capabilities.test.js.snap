// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests the rendering of Capabilities 1`] = `
<div
  id="capabilities-page"
>
  <div
    className="capabilities-container background-light-translucent border-radius-15 box-shadow"
  >
    <h1
      className="heading-text main-heading foreground-back capital-text center-text"
    >
      Capabilities
    </h1>
    <div
      className="capabilities-links"
    >
      <a
        className="capability-link"
        href="#integrity"
      >
        <div
          className="img-container background-1"
        >
          <img
            alt=""
            src="integrity.png"
          />
        </div>
        <h1
          className="heading-text foreground-accent1 capital-text center-text"
        >
          Voting Integrity
        </h1>
      </a>
      <a
        className="capability-link"
        href="#security"
      >
        <div
          className="img-container background-1"
        >
          <img
            alt=""
            src="security.png"
          />
        </div>
        <h1
          className="heading-text foreground-accent1 capital-text center-text"
        >
          Security
        </h1>
      </a>
      <a
        className="capability-link"
        href="#admin"
      >
        <div
          className="img-container background-1"
        >
          <img
            alt=""
            src="admin.png"
          />
        </div>
        <h1
          className="heading-text foreground-accent1 capital-text center-text"
        >
          Admin Experience
        </h1>
      </a>
      <a
        className="capability-link"
        href="#voter"
      >
        <div
          className="img-container background-1"
        >
          <img
            alt=""
            src="voter.png"
          />
        </div>
        <h1
          className="heading-text foreground-accent1 capital-text center-text"
        >
          Voter Experience
        </h1>
      </a>
    </div>
  </div>
  <div
    className="capabilities-container border-radius-15 box-shadow capability foreground-light background-1-translucent"
    id="integrity"
  >
    <div
      className="capability-headings"
    >
      <h1
        className="heading-text capital-text"
      >
        Voting Integrity
      </h1>
      <h2
        className="subheading-text"
      >
        Trust the results of your votes and elections
      </h2>
    </div>
    <ul
      className="capability-points"
    >
      <li>
        <div
          className="subheading-text"
        >
          One-to-One Voting
        </div>
        <p
          className="normal-text"
        >
          Ensure voters are not able to cast multiple votes on the same ballot
        </p>
      </li>
      <li>
        <div
          className="subheading-text"
        >
          Voter Authentication
        </div>
        <p
          className="normal-text"
        >
          The voters are authenticated using Machine learning techniques like face detection and face recognition
        </p>
      </li>
    </ul>
  </div>
  <br />
  <br />
  <div
    className="capabilities-container border-radius-15 box-shadow capability foreground-back background-light-translucent"
    id="security"
  >
    <div
      className="capability-headings"
    >
      <h1
        className="heading-text capital-text"
      >
        Security
      </h1>
      <h2
        className="subheading-text"
      >
        Ensure the election's security and peace of mind.
      </h2>
    </div>
    <ul
      className="capability-points"
    >
      <li>
        <div
          className="subheading-text"
        >
          Anonymity of votes
        </div>
        <p
          className="normal-text"
        >
          Even the admin will not be able to see who voted for whom. Thus maintaining the anonymity of votes.
        </p>
      </li>
      <li>
        <div
          className="subheading-text"
        >
          Voter Login
        </div>
        <p
          className="normal-text"
        >
          Face detection technique is used to make sure only one person logs in while voting. Face recognition is also carried out to ensure that the person casting the vote and the one logged in are the same 
        </p>
      </li>
    </ul>
  </div>
  <br />
  <br />
  <div
    className="capabilities-container border-radius-15 box-shadow capability foreground-light background-1-translucent"
    id="admin"
  >
    <div
      className="capability-headings"
    >
      <h1
        className="heading-text capital-text"
      >
        Admin Experience
      </h1>
      <h2
        className="subheading-text"
      >
        Manage the voting process with ease
      </h2>
    </div>
    <ul
      className="capability-points"
    >
      <li>
        <div
          className="subheading-text"
        >
          Voting Platform
        </div>
        <p
          className="normal-text"
        >
          Gain control over your voting events. See the results analysis and everything at one place
        </p>
      </li>
      <li>
        <div
          className="subheading-text"
        >
          Vote Management Services
        </div>
        <p
          className="normal-text"
        >
          The admin can create groups of voters for different elections like club elections, student council elections, etc.
        </p>
      </li>
      <li>
        <div
          className="subheading-text"
        >
          Voter Upload Modules
        </div>
        <p
          className="normal-text"
        >
          Our voter list upload modules let you manage the list of eligible voters, their details, and login information.
        </p>
      </li>
      <li>
        <div
          className="subheading-text"
        >
          Vote Scheduling Options
        </div>
        <p
          className="normal-text"
        >
          Set elections to open and close at a preferred date and time
        </p>
      </li>
    </ul>
  </div>
  <br />
  <br />
  <div
    className="capabilities-container border-radius-15 box-shadow capability foreground-back background-light-translucent"
    id="voter"
  >
    <div
      className="capability-headings"
    >
      <h1
        className="heading-text capital-text"
      >
        Voter Experience
      </h1>
      <h2
        className="subheading-text"
      >
        Offer voters a straightforward, pleasant experience
      </h2>
    </div>
    <ul
      className="capability-points"
    >
      <li>
        <div
          className="subheading-text"
        >
          Voter Notifications
        </div>
        <p
          className="normal-text"
        >
          Even the admin will not be able to see who voted for whom. Thus maintaining the anonymity of votes.
        </p>
      </li>
      <li>
        <div
          className="subheading-text"
        >
          Effective and User Friendly UI
        </div>
        <p
          className="normal-text"
        >
          Providing voters an alternate login option using their college ID cards which is a unique-to-the-voter link that bypasses the traditional login page.
        </p>
      </li>
      <li>
        <div
          className="subheading-text"
        >
          Candidate Manifestos
        </div>
        <p
          className="normal-text"
        >
          There is an option to view the manifestos of all the candidates so that voters can make a more informed decision
        </p>
      </li>
      <li>
        <div
          className="subheading-text"
        >
          Social Media
        </div>
        <p
          className="normal-text"
        >
          In order to increase the voter turnout,we provide an option for the people who voted to upload a picture on their social media
        </p>
      </li>
    </ul>
  </div>
  <br />
  <br />
  <div
    style={
      Object {
        "opacity": "0",
      }
    }
  >
    .
  </div>
</div>
`;
